### PR TITLE
pychess: init at 1.0.5

### PR DESCRIPTION
**Pull request by lgbishop**
## Description of changes

PyChess is an advanced GTK chess client originally developed for GNOME. The goal of the project is to provide a free and feature-rich chess client for Linux following the GNOME Human Interface Guidelines.

### Some notes:

- All functionality has been tested and seems to be in fully working order (on x86_64 Linux at least). This includes sound, graphics, engines, online play, saved games, and chess databases.
  - This is despite the (seemingly benign) error message thrown immediately upon running the program:
`ERROR: gst_player requires pygobject to be installed.`

- The `preBuild` commands are required when using the source (tagged or otherwise). One could alternatively use the release `url = "https://github.com/pychess/${pname}/releases/download/${version}/${pname}-${version}.tar.gz"` (via `fetchurl`) which contains some pre-built assets.
  - Both methods require manual copying/moving/symlinking of some assets as accomplished in `postInstall`.

- Compatible chess engines like Stockfish can optionally be installed independently and used in the game (provided they are on $PATH, or alternatively imported manually in-game).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/py/pychess/package.nix [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+  preFixup = ''
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+  '';
+
+  preBuild = ''
+    export HOME=$(mktemp -d)
+    export PYTHONPATH=./lib:$PYTHONPATH
+    ${python3Packages.python.interpreter} pgn2ecodb.py
+    ${python3Packages.python.interpreter} create_theme_preview.py
```
Comment (Aleksanaa) [inline pkgs/by-name/py/pychess/package.nix:56]:
```suggestion
    python pgn2ecodb.py
    python create_theme_preview.py
```

You won't want that `buildPackages...` stuff I suppose.

File: pkgs/by-name/py/pychess/package.nix [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  '';
+
+  postInstall = ''
+    cp -r $out/share/pychess/* $out/lib/python*/
```
Comment (Aleksanaa) [inline pkgs/by-name/py/pychess/package.nix:60]: Is this needed to be able to execute the app?

File: pkgs/by-name/py/pychess/package.nix [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+python3Packages.buildPythonApplication rec {
+  pname = "pychess";
+  version = "1.0.5";
+
+  src = fetchFromGitHub {
+    owner = "pychess";
+    repo = "pychess";
+    rev = "${version}";
+    sha256 = "sha256-hxc+vYvCeiM0+oOu1peI9qkZg5PeIsDMCiydJQAuzOk=";
```
Comment (Aleksanaa) [inline pkgs/by-name/py/pychess/package.nix:21]:
```suggestion
    hash = "sha256-hxc+vYvCeiM0+oOu1peI9qkZg5PeIsDMCiydJQAuzOk=";
```

File: pkgs/by-name/py/pychess/package.nix [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+  postInstall = ''
+    cp -r $out/share/pychess/* $out/lib/python*/
+  '';
+
+  # No tests available.
+  doCheck = false;
+
+  meta = {
+    description = "An advanced GTK chess client written in Python";
```
Comment (Aleksanaa) [inline pkgs/by-name/py/pychess/package.nix:67]:
```suggestion
    description = "Advanced GTK chess client written in Python";
```

File: pkgs/by-name/py/pychess/package.nix (hunk 2 above)
Comment (lgbishop) [reply pkgs/by-name/py/pychess/package.nix:60]: I do believe so. Without the assets copied across, the app fails to execute and throws the following:
```
Warning: eco.db not found, please run pgn2ecodb.sh
Traceback (most recent call last):
  File "/nix/store/q3hqdlh51r5jp9fqzjg59k0l1i4wj1q3-pychess-1.0.5/bin/.pychess-wrapped", line 237, in <module>
    from pychess.Main import PyChess
  File "/nix/store/q3hqdlh51r5jp9fqzjg59k0l1i4wj1q3-pychess-1.0.5/lib/python3.12/site-packages/pychess/Main.py", line 22, in <module>
    from pychess.widgets import enginesDialog, newGameDialog
  File "/nix/store/q3hqdlh51r5jp9fqzjg59k0l1i4wj1q3-pychess-1.0.5/lib/python3.12/site-packages/pychess/widgets/enginesDialog.py", line 14, in <module>
    from pychess.widgets import newGameDialog
  File "/nix/store/q3hqdlh51r5jp9fqzjg59k0l1i4wj1q3-pychess-1.0.5/lib/python3.12/site-packages/pychess/widgets/newGameDialog.py", line 70, in <module>
    from pychess.widgets.BoardControl import BoardControl
  File "/nix/store/q3hqdlh51r5jp9fqzjg59k0l1i4wj1q3-pychess-1.0.5/lib/python3.12/site-packages/pychess/widgets/BoardControl.py", line 41, in <module>
    from . import preferencesDialog
  File "/nix/store/q3hqdlh51r5jp9fqzjg59k0l1i4wj1q3-pychess-1.0.5/lib/python3.12/site-packages/pychess/widgets/preferencesDialog.py", line 34, in <module>
    from pychess.gfx import Pieces
  File "/nix/store/q3hqdlh51r5jp9fqzjg59k0l1i4wj1q3-pychess-1.0.5/lib/python3.12/site-packages/pychess/gfx/Pieces.py", line 128, in <module>
    makruk_svg_pieces = get_svg_pieces("makruk")
                        ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/q3hqdlh51r5jp9fqzjg59k0l1i4wj1q3-pychess-1.0.5/lib/python3.12/site-packages/pychess/gfx/Pieces.py", line 116, in get_svg_pieces
    rsvg_handles[c][p] = Rsvg.Handle.new_from_file(
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^
gi.repository.GLib.GError: rsvg-error-quark: I/O error: Error opening file /nix/store/q3hqdlh51r5jp9fqzjg59k0l1i4wj1q3-pychess-1.0.5/lib/python3.12/pieces/makruk/wp.svg: No such file or directory (0)
```
(In addition to the seemingly benign `ERROR: gst_player requires pygobject to be installed.`)

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**